tests/test_github_repo_request_local.py

```diff
@@ -1,128 +1,82 @@
-
-    # URLs with query parameters or fragments
-    assert (
-        get_base_repo_url("https://github.com/getdnsapi/stubby.git#readme")
-        == "https://github.com/getdnsapi/stubby.git"
-    )
-
-    assert (
-        get_base_repo_url("https://github.com/getdnsapi/stubby.git?branch=main")
-        == "https://github.com/getdnsapi/stubby.git"
-    )
-    # Writing test for direct_path_platforms
-    #         "gitlab.com",
-    #         "gitlab.torproject.org",
-    #         "codeberg.org",
-    #         "framagit.org",
-    #         "hydrillabugs.koszko.org",
-    #         "git.replicant.us",
-    #         "gerrit.osmocom.org",
-    #         "git.taler.net"
-
-    assert (
-        get_base_repo_url("https://gitlab.com/technostructures/kazarma/kazarma")
-        == "https://gitlab.com/technostructures/kazarma/kazarma"
-    )
-
-    assert (
-        get_base_repo_url("https://gitlab.torproject.org/tpo/network-heal")
-        == "https://gitlab.torproject.org/tpo/network-heal"
-    )
-
-    assert (
-        get_base_repo_url("https://codeberg.org/interpeer")
-        == "https://codeberg.org/interpeer"
-    )
-
-    assert (
-        get_base_repo_url("https://codeberg.org/openEngiadina/geopub")
-        == "https://codeberg.org/openEngiadina/geopub"
-    )
-
-    assert (
-        get_base_repo_url("https://codeberg.org/librEDA")
-        == "https://codeberg.org/librEDA"
-    )
-
-    assert (
-        get_base_repo_url("https://framagit.org/incommon.cc")
-        == "https://framagit.org/incommon.cc"
-    )
-
-    assert (
-        get_base_repo_url(
-            "https://hydrillabugs.koszko.org/projects/haketilo/repository"
-        )
-        == "https://hydrillabugs.koszko.org/projects/haketilo/repository"
-    )
-
-    assert (
-        get_base_repo_url("https://git.replicant.us/replicant")
-        == "https://git.replicant.us/replicant"
-    )
-
-    assert (
-        get_base_repo_url("https://gerrit.osmocom.org/plugins/gitiles/ims")
-        == "https://gerrit.osmocom.org/plugins/gitiles/ims"
-    )
-
-    assert get_base_repo_url("https://git.taler.net/taler-ios.git") == (
-        "https://git.taler.net/taler-ios.git"
-    )
+    # Test data setup
+    data = pd.DataFrame(
+        {
+            "repourl": [
+                None,  # Test for None input
+                "https://github.com/getdnsapi/stubby",
+                "https://github.com/namecoin",
+                "https://github.com/osresearch/heads/issues/540",
+                "https://git.sr.ht/~dvn/boxen",
+                "https://redmine.replicant.us/projects/replicant/wiki/"
+                "Tasks_funding#Graphics-acceleration",
+                "https://github.com/tdf/odftoolkit.git",
+                "https://git.savannah.gnu.org/git/mes.git",
+                "https://git.taler.net/git/libtalerutil.git",
+                "https://git.taler.net/taler-ios.git",
+                "just-a-string",
+                "http:///a-bad-url.com",
```

Comment:
It is a bad URL because of triple slash (///). 
The conversion of `http` to `https` is done in the line 351 in the script `initial_data_preparation.py`. Added a comment to the issue #72 to add test cases for this.